test/catalog/constraints_test.cpp

```diff
@@ -55,54 +75,848 @@ TEST_F(ConstraintsTests, NOTNULLTest) {
+
+  // Insert some records
+  txn = txn_manager.BeginTransaction();
+  LOG_INFO("================================================");
+  LOG_INFO("============Starting to insert records==========");
+
+  std::string q1 = "INSERT INTO test_table VALUES (1, 10, NULL);";
+
+  std::unique_ptr<Statement> statement;
+  statement.reset(new Statement("INSERT", q1));
+  auto& peloton_parser = parser::PostgresParser::GetInstance();
+  auto insert_stmt = peloton_parser.BuildParseTree(q1);
+
+  statement->SetPlanTree(optimizer.BuildPelotonPlanTree(insert_stmt));
+  std::vector<type::Value> params;
+  std::vector<StatementResult> result;
+  LOG_INFO("Executing plan...\n%s",
+           planner::PlanUtil::GetInfo(statement->GetPlanTree().get()).c_str());
+
+  std::vector<int> result_format;
+  result_format =
+    std::move(std::vector<int>(statement->GetTupleDescriptor().size(), 0));
+  bridge::peloton_status status = traffic_cop.ExecuteStatementPlan(
+    statement->GetPlanTree().get(), params, result, result_format);
+  LOG_INFO("Statement executed. Result: %s",
+           ResultTypeToString(status.m_result).c_str());
+  LOG_INFO("Tuple inserted!");
+
+  txn_manager.CommitTransaction(txn);
+
+  // Do a select query to check the results
+  txn = txn_manager.BeginTransaction();
+  // Do a select to get the original values
+  std::unique_ptr<Statement> statement1;
+  statement1.reset(new Statement("SELECT", "SELECT * FROM test_table LIMIT 1;"));
+  auto select_stmt =
+    peloton_parser.BuildParseTree("SELECT * FROM test_table LIMIT 1;");
+
+  statement1->SetPlanTree(optimizer.BuildPelotonPlanTree(select_stmt));
+  std::vector<type::Value> params1;
+  std::vector<StatementResult> result1;
+  // bridge::PlanExecutor::PrintPlan(statement->GetPlanTree().get(), "Plan");
+  std::vector<int> result_format1;
+  auto tuple_descriptor = traffic_cop.GenerateTupleDescriptor(
+    select_stmt->GetStatement(0));
+  result_format1 = std::move(std::vector<int>(tuple_descriptor.size(), 0));
+  UNUSED_ATTRIBUTE bridge::peloton_status status1 =
+    traffic_cop.ExecuteStatementPlan(statement1->GetPlanTree().get(), params1,
+                                      result1, result_format1);
+  LOG_INFO("Statement executed. Result: %s",
+           ResultTypeToString(status1.m_result).c_str());
+  txn_manager.CommitTransaction(txn);
+
+  // Check the results
+  // Todo: having trouble checking the tuple values
+  std::string s1, s2, col3val;
+
+  for (unsigned int i = 0; i < result1.size(); i++) {
+    StatementResult r = result1[i];
+    std::string ss1, ss2;
+
+    for (unsigned char c : r.first) {
+      ss1 += c;
+    }
+
+    for (unsigned char c : r.second) {
+      ss2 += c;
+    }
+
+    s1 += ss1 + " ";
+    s2 += ss2 + " ";
+
+    if (i == result1.size() - 1) {
+      col3val += ss2;
+    }
+  }
+
+  LOG_INFO("SELECT result from 1 : %s", s1.c_str());
+  LOG_INFO("SELECT result from 2 : %s", s2.c_str());
+
+  EXPECT_EQ(DEFAULT_VALUE, std::stoi(col3val));
```

Comment:
Not sure what this test exactly tries to achieve. But I guess there should be other ways to get the tuples values to check for quality.